concrete/src/File/Search/SearchProvider.php

```diff
@@ -65,12 +68,26 @@ public function getItemList()
         return $list;
     }
 
+    public function getItemsPerPage()
+    {
+        $searchRequest = new StickyRequest('file_manager_folder');
+        $searchParams = $searchRequest->getSearchRequest();
+        $node = empty($searchParams['folder']) ? null : Node::getByID($searchParams['folder']);
+        if ($node instanceof SearchPreset) {
+            $searchObj = $node->getSavedSearchObject();
+            return $searchObj->getQuery()->getItemsPerPage();
+        } else {
+            return parent::getItemsPerPage();
```

Comment:
Just as a nitpick, this would be easier to read if you used an "Early return" if $node is not an instance you recognize and no conditional for the successful return:

```php
if (!$node instanceof SearchPreset) {
    return parent::getItemsPerPage();
}

$searchObj = $node->getSavedSearchObject();
return $searchObj->getQuery()->getItemsPerPage();
```

vs
```php
if ($node instanceof SearchPreset) {
    $searchObj = $node->getSavedSearchObject();
    return $searchObj->getQuery()->getItemsPerPage();
} else {
    return parent::getItemsPerPage();
}
```

This has the added benefit of making it clear which return is the expected return in most cases

Reply:
👍